module_utils/network_lsr/argument_validator.py

```diff
@@ -633,6 +704,177 @@ def _validate_post(self, value, name, result):
         return result
 
 
+class ArgValidatorIPRoutingRules(ArgValidatorDict):
+    def __init__(self, name, family=None, required=False):
+        ArgValidatorDict.__init__(
+            self,
+            name,
+            required,
+            nested=[
+                # a valid priority ranges from 0 to 0xFFFFFFFF, "-1" is also allowed
+                # to reset the priority
```

Comment:
I have not fully tested resetting the priority, I got hint from here, https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/blob/main/src/libnm-core-impl/nm-setting-ip-config.c#L1812.

Reply:
`nm_ip_routing_rule_set_priority()` is a function of libnm. In libnm, `NMIPRoutingRule` is a relatively simple stucture that contains information about rules. It has setters and getters. Initially, when you create a new `NMIPRoutingRule` instance, it has no priority set. So there is a setter to set a priority, but the setter also allows to reset the unset priority.

Of course, a `NMIPRoutingRule` instance without a priority is not valid. The ability to reset the priority is merely for completeness, so that the setters allow you to create a `NMIPRoutingRule` in any configuration -- even an invalid one.

Having that part of the playbook API where users could create an invalid rule (without a priority) is not right. In fact, the priority is a mandatory parameter in the playbook, they an already create an invalid rule by omitting it.